### ISSUE TITLE
[Feature] Allow specifying job id as input to enqueue/schedule 

### ISSUE THREAD

**Issue by johanhaleby** [jobrunr/jobrunr#47]
When falling asleep yesterday I came to think of an important thing that I don't think JobRunr currently supports, and that is idempotency of job scheduling/enqueueing. 

For example, in event-based systems, it's common that I want to schedule something after an event has occurred. Take for example the order case. Let's say you an order was placed, i.e. an `OrderWasPlaced` event has been published. As a consequence of this I want to schedule a reminder to be sent if the order has not been paid in 1 week. Thus I'd like to create a [policy](https://occurrent.org/documentation#policy) that listens to `OrderWasPlaced` and schedules a future action (pseudo-code since I don't have the API in front of me :)):

```java
OrderWasPlaced e = ..
BackgroundJob.schedule(() -> Order.sendReminder(..), e.getPlacedAt().plus(1, WEEK));
```

But if the application crash after the call to `BackgroundJob.schedule(..)` but _before_ the method returns and ack's that the `OrderWasPlaced` has been handled correctly, then two jobs will be registered since the method will be retried by the broker infrastructure (RabbitMQ, Kafka etc).

Instead, I would like the client to also be able to generate a unique UUID. For example:

```java
OrderWasPlaced e = ..
BackgroundJob.schedule(settings().jobId(e.getEventUUID()), () -> Order.sendReminder(..), e.getPlacedAt().plus(1, WEEK));
```

`settings()` could also host more options, such as the ability to specify a `parallel` property as discussed in issue #43. 

Note that it would be benefitial if `settings` could be defined before the lamda expression if you take Kotlin friendliness into account since in Kotlin the last lambda expression can be moved outside the parenthesis: 

```kotin
BackgroundJob.enqueue(settings().jobId(UUID.randomUUID()) {
    Application.doStuff(...)
}
```

Unfortunately, this won't work for the `schedule` method since it defines `OffsetDateTime` and `ZoneDateTime` as the last parameter. It's a minor thing, but maybe it's worth keeping in mind in the future :)

**Comment by rdehuyss** [comment]
Related to #17

**Comment by rdehuyss** [comment]
Solved with the release of JobRunr v2.0.0.
